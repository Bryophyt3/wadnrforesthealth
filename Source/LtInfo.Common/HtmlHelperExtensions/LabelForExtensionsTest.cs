using System.Web;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace LtInfo.Common.HtmlHelperExtensions
{
    [TestFixture]
    public class LabelForExtensionsTest
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateLabelWithFieldDefinitionForNotYetDefinedFieldDefinitionTest()
        {
            var result = LabelForExtensions.LabelWithFieldDefinitionForImpl("My Field",
                "myField",
                null,
                "/FieldDefinition/FieldDefinitionDetails/1",
                300,
                LabelForExtensions.DisplayStyle.HelpIconWithLabel, false);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateLabelWithFieldDefinitionForDefinedFieldDefinitionTest()
        {
            var fieldDefinitionData = new TestFieldDefinitionData(-1, "My field's definition");
            var result = LabelForExtensions.LabelWithFieldDefinitionForImpl("My Field",
                "myField",
                fieldDefinitionData,
                "/FieldDefinition/FieldDefinitionDetails/1",
                300,
                LabelForExtensions.DisplayStyle.HelpIconWithLabel, false);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateLabelWithFieldDefinitionForDefinedFieldDefinitionForGridHeaderTest()
        {
            var fieldDefinitionData = new TestFieldDefinitionData(-1, "My field's definition");
            var result = LabelForExtensions.LabelWithFieldDefinitionForImpl("My Field",
                "myField",
                fieldDefinitionData,
                "/FieldDefinition/FieldDefinitionDetails/1",
                300,
                LabelForExtensions.DisplayStyle.AsGridHeader, false);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateLabelWithFieldDefinitionForDefinedFieldDefinitionForHelpIconOnlyTest()
        {
            var fieldDefinitionData = new TestFieldDefinitionData(-1, "My field's definition");
            var result = LabelForExtensions.LabelWithFieldDefinitionForImpl("My Field",
                "myField",
                fieldDefinitionData,
                "/FieldDefinition/FieldDefinitionDetails/1",
                300,
                LabelForExtensions.DisplayStyle.HelpIconOnly, false);
            Approvals.Verify(result);
        }
    }

    public class TestFieldDefinitionData : IFieldDefinitionData
    {
        public HtmlString FieldDefinitionDataValueHtmlString { get; private set; }

        public TestFieldDefinitionData(int id, string definition)
        {
            FieldDefinitionDataValueHtmlString = new HtmlString(definition);   
        }
    }
}