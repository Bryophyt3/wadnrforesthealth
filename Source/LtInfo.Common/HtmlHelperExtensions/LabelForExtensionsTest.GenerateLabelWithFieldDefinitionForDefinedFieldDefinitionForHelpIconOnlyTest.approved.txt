<span class="helpicon glyphicon glyphicon-question-sign helpiconGridBlue" onmouseover="Corral.Views.Methods.addHelpTooltipPopup(this, &#39;My Field&#39;, &#39;\/FieldDefinition\/FieldDefinitionDetails\/1&#39;, 300)" title="Click to get help on My Field"></span>